x/fswap/client/cli/tx.go

```diff
@@ -22,6 +22,10 @@ const (
 	FlagSwapRate            = "swap-rate"
 )
 
+type ToDenomMeta struct {
+	Metadata bank.Metadata `json:"metadata"`
+}
+
```

Comment:
In general, It's not desirable to make it public for test.
Is there any way to test without exposing it?

Reply:
Do you have some ideas?

Reply:
Please wait, let me check.

Reply:
First you need to revert `ToDenomMeta` to private, then you can change `TestMsgSetSwap` code like below.

```go
func (s *IntegrationTestSuite) TestMsgSetSwap() {
	val := s.network.Validators[0]
	clientCtx := val.ClientCtx
	// avoid printing as yaml from CLI command
	clientCtx.OutputFormat = jsonOutputFormat

	denomMeta := struct {
		Metadata bank.Metadata `json:"metadata"`
	}{
		Metadata: s.toDenom,
	}
```

Reply:
Using the same tag(`json:"metadata"`) is enough to run this test.

Reply:
Yes, but this way defining the same structure in two places indeed increases the maintenance cost, as any changes to the structure need to be manually synchronized across both definitions. Is this acceptable?

Reply:
I can't come up another solution for this but exposing code for testing is a lot worse.

Reply:
I also have same opinion. Exposing implementation detail as public API is usually bad idea. Because, after exposing it, we cannot change them or the change must be treated as API breaking change.